lib/index.js

```diff
@@ -0,0 +1,56 @@
+// Knex.js  0.8.0
+// --------------
+//     (c) 2014 Tim Griesser
+//     Knex may be freely distributed under the MIT license.
+//     For details and documentation:
+//     http://knexjs.org
+
+'use strict';
+
+var Raw            = require('./raw')
+var warn           = require('./helpers').warn
+var Client         = require('./client')
+
+var makeClient     = require('./util/make-client')
+var makeKnex       = require('./util/make-knex')
+var assign         = require('lodash/object/assign')
+
+function Knex(config) {
+  if (typeof config === 'string') {
+    return new Knex(assign(parseUrl(config), arguments[2]))
+  }
+  var Dialect;
+  if (arguments.length === 0 || (!config.client && !config.dialect)) {
+    Dialect = makeClient(Client)
+  } else {
+    var clientName = config.client || config.dialect
+    Dialect = makeClient(require('./dialects/' + (aliases[clientName] || clientName) + '/index.js'))
```

Comment:
So I ended up making a specific [websql build](https://github.com/tgriesser/knex/blob/3e47fafbe8ecc35e2e163c2ed87e37264ce700da/browser/websql.js) with webpack which keeps all `node_modules` [external from the build](https://github.com/tgriesser/knex/blob/3e47fafbe8ecc35e2e163c2ed87e37264ce700da/webpack.config.js#L42-L44) but does a bit of cleanup to ensure unnecessary files/dialects [aren't loaded](https://github.com/tgriesser/knex/blob/3e47fafbe8ecc35e2e163c2ed87e37264ce700da/webpack.config.js#L3-L15) and then I'm just pointing "browser" at that file.

Does this seem reasonable?